### ISSUE TITLE
Incorrect test environment when using storeConfigInMeta: false

### ISSUE THREAD

**Issue by mmun** [ember-cli/ember-cli#4535]
When using `storeConfigInMeta: false` the testing environment is `"development"` instead of `"test"`. Steps to reproduce:
- Set `storeConfigInMeta: false`
- `ember serve`
- Visit http://localhost:4200/tests/index.html
- In the console, `require('my-app/config/environment').default.environment` is `development` instead of `test`.

In acceptance tests, this means that the router's `locationType` is not set to `"none"` which is really annoying to work with.

**Comment by mmun** [comment]
A workaround is to manually setup the environment in `tests/index.html` after the asset script tags:

```html
    ...

    <script>
      var config = require('my-app/config/environment')['default'];
      config.environment = "test";
      config.locationType = "none";
    </script>

    {{content-for 'body-footer'}}
    {{content-for 'test-body-footer'}}
```

**Comment by rwjblue** [comment]
We can make this work by default with `storeConfigInMeta: false` by storing the result of `config/environment.js` in two modules (`config/environment/development` and `config/environment/test`), and `define`ing the `config/environment` by checking `storeConfigInMeta` when generating the `test-support.js` file.

**Comment by amk221** [comment]
I needed `storeConfigInMeta: false` to avoid this error message:

    Could not read config from meta tag with name


(Because I don't use ember-cli's server for development, but rather just as a build/test tool)

A side affect of this was that running `ember test` ran in 'development' instead of 'test'. Because tests/index.html obviously didn't know the config.

A solution for me was to do add this to `ember-cli-build.js`: 

    storeConfigInMeta: EmberApp.env() === 'test'

Best of both worlds

**Comment by stefanpenner** [comment]
related: https://github.com/ember-cli/ember-cli/issues/3299

**Comment by raido** [comment]
I got into this trap today. I have been using Ember for years and until recently I used Ember.env() checks for storeConfigInMeta. Now i decided to go simply with **false** on my new app and hours went to debugging why config is not correct.

**Comment by rwjblue** [comment]
Fixed by #7528
